lxd/db_images.go

```diff
@@ -105,8 +105,17 @@ func dbImageSourceGet(db *sql.DB, imageId int) (int, api.ImageSource, error) {
 
 }
 
-// dbImageGet gets an ImageBaseInfo object from the database.
-// The argument fingerprint will be queried with a LIKE query, means you can
+// Whether an image with the given fingerprint exists.
+func dbImageExists(db *sql.DB, fingerprint string) (exists bool, err error) {
```

Comment:
I get the public vs private argument but your function is subject to db errors just like `dbImageGet()`. Can't we just have something along the lines of:

```
func dbImageExists(db *sql.DB, fingerprint string) (bool, error) {
	query := "SELECT COUNT(*) > 0 FROM images WHERE fingerprint=?"
	inargs := []interface{}{fingerprint}
	outargs := []interface{}{&exists}

	err = dbQueryRowScan(db, query, inargs, outargs)
	if err != nil {
		if err != sql.ErrNoRows {
			return false, err
		}
		return false, nil
	}
	return true, nil
}
```